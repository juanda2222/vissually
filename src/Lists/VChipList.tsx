import React from 'react'
import VChip from "../Display/VChip"

function VChipList ({
  //functional
  onDelete,
  onClick,
  getListLabel,
  list,
}: ChipListProps ){

  // style the children:
  const wrapped_list = list ? list : []
  const list_component =  wrapped_list.map((item, index) => {

    let label = ""
    // depending on the list structure get the label:
    if (typeof getListLabel === 'function') {
      label = getListLabel(item)
      // a plain text list:
    } else {
      label = item
    }

    return (
      <VChip
        style={{margin:"2px"}}
        key={label}
        label={label}
        onClick={() => {
          if (typeof onClick == "function") { onClick({ label: label, index: index }) }
        }}
        onDelete={() => {
          if (typeof onDelete == "function") { onDelete({ label: label, index: index }) }
        }}
      />
    )
  })

  return (
    <div
      style={{
        flex:1,
        display: "flex",
        flexWrap: "wrap",
        justifyContent:"center",
        alignItems: "flex-start",
      }}
    >
    {list_component}
  </div>)
}


export default VChipList