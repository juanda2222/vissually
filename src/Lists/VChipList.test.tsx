
import React from "react"
import renderer from 'react-test-renderer';
import VChipList from './VChipList'

describe('VChipList', () => {

  test('VChipList declaration has no errors', () => {
    expect(VChipList).toBeTruthy()
  })

  test("Default render. no list given", () => { 
    const vChipListSnap = renderer.create(<VChipList list={["1", "2"]}/>).toJSON();
    console.log(vChipListSnap)
    expect(vChipListSnap).toMatchSnapshot();
  })

  test("Rendering with props. Snapshot test", () => { 

    const vChipListSnap = renderer.create(
      <VChipList
        list={["some_thing", "some_name", "some_name2", "other name", "whatever list"]}
        onClick={(i: number, text: string) => {
          console.log("clicked! index:", i.toString(10), " Text: ", text)
        }}
        onDelete={(i: number, text: string) => {
          console.log("Deleted! index:", i.toString(10), " Text: ", text)
        }}
      />
    ).toJSON();

    console.log(vChipListSnap)
    expect(vChipListSnap).toMatchSnapshot();
  })

  test("Rendering with props. Snapshot test (With getLabel property)", () => { 

    const vChipListSnap = renderer.create(
      <VChipList
        list={["some_thing", "some_name", "some_name2", "other name", "whatever list"]}
        getListLabel={(element) => element}
        onClick={(i: number, text: string) => {
          console.log("clicked! index:", i.toString(10), " Text: ", text)
        }}
        onDelete={(i: number, text: string) => {
          console.log("Deleted! index:", i.toString(10), " Text: ", text)
        }}
      />
    ).toJSON();

    console.log(vChipListSnap)
    expect(vChipListSnap).toMatchSnapshot();
  })


})
