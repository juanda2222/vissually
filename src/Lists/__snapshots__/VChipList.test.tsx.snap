// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VChipList Default render. no list given 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flex": 1,
      "flexWrap": "wrap",
      "justifyContent": "center",
    }
  }
>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      1
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      2
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`VChipList Rendering with props. Snapshot test (With getLabel property) 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flex": 1,
      "flexWrap": "wrap",
      "justifyContent": "center",
    }
  }
>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      some_thing
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      some_name
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      some_name2
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      other name
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      whatever list
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`VChipList Rendering with props. Snapshot test 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flex": 1,
      "flexWrap": "wrap",
      "justifyContent": "center",
    }
  }
>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      some_thing
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      some_name
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      some_name2
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      other name
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
  <div
    className="sc-AxjAm dkuJXE container "
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
      }
    }
  >
    <span>
      whatever list
    </span>
    <div
      className="sc-AxirZ kWaCOc delete_button"
      onClick={[Function]}
    >
      <svg
        height="24"
        style={
          Object {
            "fill": "rgb(0, 0, 0)",
            "height": "0.6em",
            "margin": "auto 6px",
            "width": "0.6em",
          }
        }
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M24 20.188l-8.315-8.209 8.2-8.282-3.697-3.697-8.212 8.318-8.31-8.203-3.666 3.666 8.321 8.24-8.206 8.313 3.666 3.666 8.237-8.318 8.285 8.203z"
        />
      </svg>
    </div>
  </div>
</div>
`;
